Reduce use of (NSString *) casting in the codebase

cdumez opened the pull request:
#### 4142580276a38fa413fb95c6edf45796ba81a622
<pre>
Reduce use of (NSString *) casting in the codebase
<a href="https://bugs.webkit.org/show_bug.cgi?id=291221">https://bugs.webkit.org/show_bug.cgi?id=291221</a>

Reviewed by Timothy Hatcher.

Use modern casting functions instead for proper casting. For converting from
a WTF::String to a NSString, use `String::createNSString()` instead, which
has the benefit of making it clear it allocates a new NSString and doesn&apos;t
return an autoreleased object.

* Source/JavaScriptCore/API/JSValue.mm:
(performPropertyOperation):
(objectToValueWithoutCopy):
(objectToValue):
* Source/WTF/wtf/cocoa/FileSystemCocoa.mm:
(WTF::FileSystemImpl::setExcludedFromBackup):
* Source/WTF/wtf/cocoa/NSURLExtras.h:
* Source/WTF/wtf/cocoa/NSURLExtras.mm:
(WTF::decodeHostName):
(WTF::encodeHostName):
* Source/WebCore/Modules/applepay/cocoa/PaymentSessionErrorCocoa.mm:
(WebCore::additionalError):
* Source/WebCore/Modules/notifications/NotificationOptionsPayloadCocoa.mm:
(WebCore::NotificationOptionsPayload::dictionaryRepresentation const):
* Source/WebCore/Modules/notifications/NotificationPayloadCocoa.mm:
(WebCore::NotificationPayload::dictionaryRepresentation const):
* Source/WebCore/accessibility/mac/AXObjectCacheMac.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
(-[WebAccessibilityObjectWrapperBase baseAccessibilitySpeechHint]):
(-[WebAccessibilityObjectWrapperBase baseAccessibilityResolvedEditingStyles]):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
(-[WebAccessibilityObjectWrapper accessibilityAttributeValue:]):
(-[WebAccessibilityObjectWrapper accessibilityAttributeValue:forParameter:]):
* Source/WebCore/crypto/cocoa/SerializedCryptoKeyWrapMac.mm:
(WebCore::createAndStoreMasterKey):
* Source/WebCore/editing/cocoa/AttributedString.mm:
(WebCore::toNSObject):
(WebCore::toNSDictionary):
(WebCore::AttributedString::nsAttributedString const):
(WebCore::extractArray):
(WebCore::extractDictionary):
* Source/WebCore/editing/cocoa/DataDetection.mm:
(WebCore::DataDetection::canBePresentedByDataDetectors):
(WebCore::resultIsURL):
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
(HTMLConverter::_processElement):
(fileWrapperForElement):
* Source/WebCore/editing/mac/EditorMac.mm:
(WebCore::Editor::plainTextFromPasteboard):
* Source/WebCore/platform/audio/ios/AudioSessionIOS.mm:
(WebCore::AudioSessionIOS::updateSpatialExperience):
* Source/WebCore/platform/cocoa/DragDataCocoa.mm:
(WebCore::DragData::asPlainText const):
* Source/WebCore/platform/cocoa/MIMETypeRegistryCocoa.mm:
(WebCore::MIMETypeRegistry::preferredExtensionForMIMEType):
* Source/WebCore/platform/cocoa/PublicSuffixStoreCocoa.mm:
(WebCore::isPublicSuffixCF):
* Source/WebCore/platform/cocoa/SerializedPlatformDataCueValue.mm:
(WebCore::SerializedPlatformDataCueValue::SerializedPlatformDataCueValue):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMInstanceFairPlayStreamingAVFObjC.mm:
(WebCore::CDMInstanceSessionFairPlayStreamingAVFObjC::loadSession):
(WebCore::CDMInstanceSessionFairPlayStreamingAVFObjC::removeSessionData):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.mm:
(WebCore::CDMSessionAVContentKeySession::releaseKeys):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::createAVAssetForURL):
* Source/WebCore/platform/ios/PasteboardIOS.mm:
(WebCore::cocoaTypeFromHTMLClipboardType):
* Source/WebCore/platform/ios/PlatformPasteboardIOS.mm:
(WebCore::addRepresentationsForPlainText):
(WebCore::PlatformPasteboard::write):
(WebCore::PlatformPasteboard::typesSafeForDOMToReadAndWrite const):
(WebCore::createItemProviderRegistrationList):
* Source/WebCore/platform/ios/PreviewConverterIOS.mm:
(WebCore::optionsWithPassword):
* Source/WebCore/platform/ios/WebItemProviderPasteboard.mm:
(classForTypeIdentifier):
(-[WebItemProviderPasteboard valuesForPasteboardType:inItemSet:]):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::Pasteboard::supportedFileUploadPasteboardTypes):
(WebCore::writeURLForTypes):
* Source/WebCore/platform/mac/PasteboardWriter.mm:
(WebCore::createPasteboardWriter):
* Source/WebCore/platform/mac/PlatformPasteboardMac.mm:
(WebCore::isFilePasteboardType):
(WebCore::PlatformPasteboard::getPathnamesForType const):
(WebCore::pasteboardMayContainFilePaths):
(WebCore::PlatformPasteboard::setStringForType):
* Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm:
(WebCore::NetworkStorageSession::httpCookiesForURL const):
(WebCore::NetworkStorageSession::setCookiePartition):
(WebCore::NetworkStorageSession::domCookiesForHost):
(WebCore::NetworkStorageSession::startListeningForCookieChangeNotifications):
(WebCore::NetworkStorageSession::stopListeningForCookieChangeNotifications):
* Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm:
(WebCore::ResourceRequest::doUpdateResourceRequest):
(WebCore::ResourceRequest::doUpdatePlatformRequest):
* Source/WebCore/platform/network/cocoa/ResourceResponseCocoa.mm:
(WebCore::ResourceResponse::initNSURLResponse const):
* Source/WebCore/platform/network/mac/ResourceErrorMac.mm:
(WebCore::createNSErrorFromResourceErrorBase):
* Source/WebCore/platform/network/mac/WebCoreResourceHandleAsOperationQueueDelegate.mm:
(-[WebCoreResourceHandleAsOperationQueueDelegate connectionDidFinishLoading:]):
* Source/WebKit/Scripts/webkit/tests/GeneratedWebKitSecureCoding.cpp:
(WebKit::vectorFromDictionary):
* Source/WebKit/Shared/Cocoa/APIObject.mm:
(API::Object::toNSObject):
* Source/WebKit/Shared/Cocoa/SandboxInitialiationParametersCocoa.mm:
(WebKit::SandboxInitializationParameters::addPathParameter):
* Source/WebKit/Shared/Cocoa/WKNSString.mm:
* Source/WebKit/Shared/JavaScriptEvaluationResult.mm:
(WebKit::JavaScriptEvaluationResult::toID):
(WebKit::JavaScriptEvaluationResult::toVariant):
* Source/WebKit/UIProcess/API/Cocoa/APIAttachmentCocoa.mm:
(API::Attachment::setFileWrapperAndUpdateContentType):
* Source/WebKit/UIProcess/API/Cocoa/WKNavigationResponse.mm:
(-[WKNavigationResponse _downloadAttribute]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommand.mm:
(-[WKWebExtensionCommand debugDescription]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView mediaType]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewTesting.mm:
(-[WKWebView _contentsOfUserInterfaceItem:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
(+[WKWebsiteDataStore _removeDataStoreWithIdentifier:completionHandler:]):
(-[WKWebsiteDataStore _getAllBackgroundFetchIdentifiers:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm:
(nullableNSString):
(-[_WKApplicationManifest rawJSON]):
(-[_WKApplicationManifest name]):
(-[_WKApplicationManifest shortName]):
(-[_WKApplicationManifest applicationDescription]):
* Source/WebKit/UIProcess/API/Cocoa/_WKNotificationData.mm:
(-[_WKNotificationData title]):
(-[_WKNotificationData lang]):
(-[_WKNotificationData body]):
(-[_WKNotificationData tag]):
(-[_WKNotificationData origin]):
(-[_WKNotificationData identifier]):
* Source/WebKit/UIProcess/API/Cocoa/_WKTargetedElementInfo.mm:
(-[_WKTargetedElementInfo debugDescription]):
* Source/WebKit/UIProcess/Cocoa/SOAuthorization/SOAuthorizationSession.mm:
(WebKit::SOAuthorizationSession::continueStartAfterDecidePolicy):
* Source/WebKit/UIProcess/Cocoa/WKShareSheet.mm:
(-[WKShareSheet presentWithParameters:inRect:completionHandler:]):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionControllerAPITestCocoa.mm:
(WebKit::WebExtensionController::testResult):
(WebKit::WebExtensionController::testEqual):
(WebKit::WebExtensionController::testLogMessage):
(WebKit::WebExtensionController::testSentMessage):
(WebKit::WebExtensionController::testAdded):
(WebKit::WebExtensionController::testStarted):
(WebKit::WebExtensionController::testFinished):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCommandCocoa.mm:
(WebKit::WebExtensionCommand::setActivationKey):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView selectedText]):
* Source/WebKit/UIProcess/ios/WKPDFView.mm:
(-[WKPDFView actionSheetAssistant:performAction:]):
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
(-[WKFullScreenViewController setLocation:]):
(-[WKFullScreenViewController loadView]):
(-[WKFullScreenViewController _showPhishingAlert]):
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::registerEditCommand):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPICommandsCocoa.mm:
(WebKit::WebExtensionContextProxy::dispatchCommandsCommandEvent):
(WebKit::WebExtensionContextProxy::dispatchCommandsChangedEvent):
* Source/WebKit/webpushd/WebClipCache.mm:
(WebPushD::WebClipCache::persist):
* Source/WebKitLegacy/ios/WebView/WebPDFViewIOS.mm:
(-[WebPDFView _checkPDFTitle]):
* Source/WebKitLegacy/ios/WebView/WebPDFViewPlaceholder.mm:
(-[WebPDFViewPlaceholder _updateTitleForDocumentIfAvailable]):
* Source/WebKitLegacy/mac/Misc/WebNSURLExtras.mm:
(-[NSString _web_decodeHostName]):
(-[NSString _web_encodeHostName]):
(-[NSString _webkit_decodeHostName]):
(-[NSString _webkit_encodeHostName]):
* Tools/TestWebKitAPI/Tests/WTF/cocoa/URLExtras.mm:
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_ParsingError)):

Canonical link: <a href="https://commits.webkit.org/293429@main">https://commits.webkit.org/293429@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/26f936a8572e219b78fc2c9e7ab1cfd5087435ec

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98880 "9 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18517 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8759 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104006 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49469 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/100925 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18812 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26967 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75276 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32406 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101884 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14296 "Found 1 new test failure: fast/forms/ios/focus-input-via-button.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89296 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55637 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/98364 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14087 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7268 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48849 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/91570 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84024 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7341 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106375 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/97510 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25977 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/18940 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84240 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26352 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85494 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83738 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21225 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28397 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6061 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/19699 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25930 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/31116 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/121126 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25750 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/33874 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29070 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27324 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionControllerAPITestCocoa.mm:
```diff
@@ -99,7 +99,7 @@
     }
 
-    RELEASE_LOG_INFO(Extensions, "Test sent message: %{public}@ %{public}@ (%{public}@:%{public}u)", (NSString *)message, (NSString *)argument, (NSString *)sourceURL, lineNumber);
+    RELEASE_LOG_INFO(Extensions, "Test sent message: %{public}@ %{public}@ (%{public}@:%{public}u)", message.createNSString().get(), (NSString *)argument, sourceURL.createNSString().get(), lineNumber);
```
```suggestion
    RELEASE_LOG_INFO(Extensions, "Test sent message: %{public}@ %{public}@ (%{public}@:%{public}u)", message.createNSString().get(), argument.createNSString().get(), sourceURL.createNSString().get(), lineNumber);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionControllerAPITestCocoa.mm:
```diff
@@ -110,7 +110,7 @@
     }
 
-    RELEASE_LOG_INFO(Extensions, "Test added: %{public}@ (%{public}@:%{public}u)", (NSString *)testName, (NSString *)sourceURL, lineNumber);
+    RELEASE_LOG_INFO(Extensions, "Test added: %{public}@ (%{public}@:%{public}u)", (NSString *)testName, sourceURL.createNSString().get(), lineNumber);
```
```suggestion
    RELEASE_LOG_INFO(Extensions, "Test added: %{public}@ (%{public}@:%{public}u)", testName.createNSString().get(), sourceURL.createNSString().get(), lineNumber);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionControllerAPITestCocoa.mm:
```diff
@@ -121,7 +121,7 @@
     }
 
-    RELEASE_LOG_INFO(Extensions, "Test started: %{public}@ (%{public}@:%{public}u)", (NSString *)testName, (NSString *)sourceURL, lineNumber);
+    RELEASE_LOG_INFO(Extensions, "Test started: %{public}@ (%{public}@:%{public}u)", (NSString *)testName, sourceURL.createNSString().get(), lineNumber);
```
```suggestion
    RELEASE_LOG_INFO(Extensions, "Test started: %{public}@ (%{public}@:%{public}u)", testName.createNSString().get(), sourceURL.createNSString().get(), lineNumber);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionControllerAPITestCocoa.mm:
```diff
@@ -139,11 +139,11 @@
 
     if (result) {
-        RELEASE_LOG_INFO(Extensions, "Test passed: %{public}@ %{public}@ (%{public}@:%{public}u)", (NSString *)testName, (NSString *)message, (NSString *)sourceURL, lineNumber);
+        RELEASE_LOG_INFO(Extensions, "Test passed: %{public}@ %{public}@ (%{public}@:%{public}u)", (NSString *)testName, message.createNSString().get(), sourceURL.createNSString().get(), lineNumber);
```
```suggestion
        RELEASE_LOG_INFO(Extensions, "Test passed: %{public}@ %{public}@ (%{public}@:%{public}u)", testName.createNSString().get(), message.createNSString().get(), sourceURL.createNSString().get(), lineNumber);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionControllerAPITestCocoa.mm:
```diff
@@ -139,11 +139,11 @@
     }
 
-    RELEASE_LOG_ERROR(Extensions, "Test failed: %{public}@ %{public}@ (%{public}@:%{public}u)", (NSString *)testName, (NSString *)message, (NSString *)sourceURL, lineNumber);
+    RELEASE_LOG_ERROR(Extensions, "Test failed: %{public}@ %{public}@ (%{public}@:%{public}u)", (NSString *)testName, message.createNSString().get(), sourceURL.createNSString().get(), lineNumber);
```
```suggestion
    RELEASE_LOG_ERROR(Extensions, "Test failed: %{public}@ %{public}@ (%{public}@:%{public}u)", testName.createNSString().get(), message.createNSString().get(), sourceURL.createNSString().get(), lineNumber);
```

xeenon commented on Source/JavaScriptCore/API/JSValue.mm:
```diff
@@ -408,8 +409,8 @@ - (NSDictionary *)toDictionary
     // If it's a NSString already, reduce indirection and just pass the NSString.
-    if ([propertyKey isKindOfClass:[NSString class]]) {
-        auto name = OpaqueJSString::tryCreate((NSString *)propertyKey);
+    if (auto* propertyKeyString = dynamic_objc_cast<NSString>(propertyKey)) {
```
```suggestion
    if (auto *propertyKeyString = dynamic_objc_cast<NSString>(propertyKey)) {
```

xeenon commented on Source/JavaScriptCore/API/JSValue.mm:
```diff
@@ -1148,8 +1149,8 @@ inline bool isNSBoolean(id object)
 
-        if ([object isKindOfClass:[NSString class]]) {
-            auto string = OpaqueJSString::tryCreate((NSString *)object);
+        if (auto* nsString = dynamic_objc_cast<NSString>(object)) {
```
```suggestion
        if (auto *nsString = dynamic_objc_cast<NSString>(object)) {
```

xeenon commented on Source/JavaScriptCore/API/JSValue.mm:
```diff
@@ -1205,8 +1206,8 @@ JSValueRef objectToValue(JSContext *context, id object)
             for (id key in [dictionary keyEnumerator]) {
-                if ([key isKindOfClass:[NSString class]]) {
-                    auto propertyName = OpaqueJSString::tryCreate((NSString *)key);
+                if (auto* keyString = dynamic_objc_cast<NSString>(key)) {
```
```suggestion
                if (auto *keyString = dynamic_objc_cast<NSString>(key)) {
```

xeenon commented on Source/WebCore/editing/cocoa/AttributedString.mm:
```diff
@@ -443,8 +443,8 @@ static MultiRepresentationHEICAttachmentData toMultiRepresentationHEICAttachment
         for (id element in array) {
-            if ([element isKindOfClass:NSString.class])
-                result.append((NSString *)element);
+            if (auto* string = dynamic_objc_cast<NSString>(element))
```
```suggestion
            if (auto *string = dynamic_objc_cast<NSString>(element))
```

xeenon commented on Source/WebCore/platform/cocoa/SerializedPlatformDataCueValue.mm:
```diff
@@ -71,8 +71,8 @@
 
-    if ([item.key isKindOfClass:NSString.class])
-        m_data->key = (NSString *)item.key;
+    if (auto* keyString = dynamic_objc_cast<NSString>(item.key))
```
```suggestion
    if (auto *keyString = dynamic_objc_cast<NSString>(item.key))
```

xeenon commented on Source/WebCore/platform/mac/PlatformPasteboardMac.mm:
```diff
@@ -157,8 +158,8 @@ static bool canWriteAllPasteboardTypes(const Vector<String>& types)
     id paths = [m_pasteboard propertyListForType:pasteboardType];
-    if ([paths isKindOfClass:[NSString class]]) {
-        pathnames.append((NSString *)paths);
+    if (auto* pathsString = dynamic_objc_cast<NSString>(paths)) {
```
```suggestion
    if (auto *pathsString = dynamic_objc_cast<NSString>(paths)) {
```

cdumez committed:
Reduce use of (NSString *) casting in the codebase
https://bugs.webkit.org/show_bug.cgi?id=291221

Reviewed by Timothy Hatcher.

Use modern casting functions instead for proper casting. For converting from
a WTF::String to a NSString, use `String::createNSString()` instead, which
has the benefit of making it clear it allocates a new NSString and doesn't
return an autoreleased object.

* Source/JavaScriptCore/API/JSValue.mm:
(performPropertyOperation):
(objectToValueWithoutCopy):
(objectToValue):
* Source/WTF/wtf/cocoa/FileSystemCocoa.mm:
(WTF::FileSystemImpl::setExcludedFromBackup):
* Source/WTF/wtf/cocoa/NSURLExtras.h:
* Source/WTF/wtf/cocoa/NSURLExtras.mm:
(WTF::decodeHostName):
(WTF::encodeHostName):
* Source/WebCore/Modules/applepay/cocoa/PaymentSessionErrorCocoa.mm:
(WebCore::additionalError):
* Source/WebCore/Modules/notifications/NotificationOptionsPayloadCocoa.mm:
(WebCore::NotificationOptionsPayload::dictionaryRepresentation const):
* Source/WebCore/Modules/notifications/NotificationPayloadCocoa.mm:
(WebCore::NotificationPayload::dictionaryRepresentation const):
* Source/WebCore/accessibility/mac/AXObjectCacheMac.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
(-[WebAccessibilityObjectWrapperBase baseAccessibilitySpeechHint]):
(-[WebAccessibilityObjectWrapperBase baseAccessibilityResolvedEditingStyles]):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
(-[WebAccessibilityObjectWrapper accessibilityAttributeValue:]):
(-[WebAccessibilityObjectWrapper accessibilityAttributeValue:forParameter:]):
* Source/WebCore/crypto/cocoa/SerializedCryptoKeyWrapMac.mm:
(WebCore::createAndStoreMasterKey):
* Source/WebCore/editing/cocoa/AttributedString.mm:
(WebCore::toNSObject):
(WebCore::toNSDictionary):
(WebCore::AttributedString::nsAttributedString const):
(WebCore::extractArray):
(WebCore::extractDictionary):
* Source/WebCore/editing/cocoa/DataDetection.mm:
(WebCore::DataDetection::canBePresentedByDataDetectors):
(WebCore::resultIsURL):
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
(HTMLConverter::_processElement):
(fileWrapperForElement):
* Source/WebCore/editing/mac/EditorMac.mm:
(WebCore::Editor::plainTextFromPasteboard):
* Source/WebCore/platform/audio/ios/AudioSessionIOS.mm:
(WebCore::AudioSessionIOS::updateSpatialExperience):
* Source/WebCore/platform/cocoa/DragDataCocoa.mm:
(WebCore::DragData::asPlainText const):
* Source/WebCore/platform/cocoa/MIMETypeRegistryCocoa.mm:
(WebCore::MIMETypeRegistry::preferredExtensionForMIMEType):
* Source/WebCore/platform/cocoa/PublicSuffixStoreCocoa.mm:
(WebCore::isPublicSuffixCF):
* Source/WebCore/platform/cocoa/SerializedPlatformDataCueValue.mm:
(WebCore::SerializedPlatformDataCueValue::SerializedPlatformDataCueValue):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMInstanceFairPlayStreamingAVFObjC.mm:
(WebCore::CDMInstanceSessionFairPlayStreamingAVFObjC::loadSession):
(WebCore::CDMInstanceSessionFairPlayStreamingAVFObjC::removeSessionData):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.mm:
(WebCore::CDMSessionAVContentKeySession::releaseKeys):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::createAVAssetForURL):
* Source/WebCore/platform/ios/PasteboardIOS.mm:
(WebCore::cocoaTypeFromHTMLClipboardType):
* Source/WebCore/platform/ios/PlatformPasteboardIOS.mm:
(WebCore::addRepresentationsForPlainText):
(WebCore::PlatformPasteboard::write):
(WebCore::PlatformPasteboard::typesSafeForDOMToReadAndWrite const):
(WebCore::createItemProviderRegistrationList):
* Source/WebCore/platform/ios/PreviewConverterIOS.mm:
(WebCore::optionsWithPassword):
* Source/WebCore/platform/ios/WebItemProviderPasteboard.mm:
(classForTypeIdentifier):
(-[WebItemProviderPasteboard valuesForPasteboardType:inItemSet:]):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::Pasteboard::supportedFileUploadPasteboardTypes):
(WebCore::writeURLForTypes):
* Source/WebCore/platform/mac/PasteboardWriter.mm:
(WebCore::createPasteboardWriter):
* Source/WebCore/platform/mac/PlatformPasteboardMac.mm:
(WebCore::isFilePasteboardType):
(WebCore::PlatformPasteboard::getPathnamesForType const):
(WebCore::pasteboardMayContainFilePaths):
(WebCore::PlatformPasteboard::setStringForType):
* Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm:
(WebCore::NetworkStorageSession::httpCookiesForURL const):
(WebCore::NetworkStorageSession::setCookiePartition):
(WebCore::NetworkStorageSession::domCookiesForHost):
(WebCore::NetworkStorageSession::startListeningForCookieChangeNotifications):
(WebCore::NetworkStorageSession::stopListeningForCookieChangeNotifications):
* Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm:
(WebCore::ResourceRequest::doUpdateResourceRequest):
(WebCore::ResourceRequest::doUpdatePlatformRequest):
* Source/WebCore/platform/network/cocoa/ResourceResponseCocoa.mm:
(WebCore::ResourceResponse::initNSURLResponse const):
* Source/WebCore/platform/network/mac/ResourceErrorMac.mm:
(WebCore::createNSErrorFromResourceErrorBase):
* Source/WebCore/platform/network/mac/WebCoreResourceHandleAsOperationQueueDelegate.mm:
(-[WebCoreResourceHandleAsOperationQueueDelegate connectionDidFinishLoading:]):
* Source/WebKit/Scripts/webkit/tests/GeneratedWebKitSecureCoding.cpp:
(WebKit::vectorFromDictionary):
* Source/WebKit/Shared/Cocoa/APIObject.mm:
(API::Object::toNSObject):
* Source/WebKit/Shared/Cocoa/SandboxInitialiationParametersCocoa.mm:
(WebKit::SandboxInitializationParameters::addPathParameter):
* Source/WebKit/Shared/Cocoa/WKNSString.mm:
* Source/WebKit/Shared/JavaScriptEvaluationResult.mm:
(WebKit::JavaScriptEvaluationResult::toID):
(WebKit::JavaScriptEvaluationResult::toVariant):
* Source/WebKit/UIProcess/API/Cocoa/APIAttachmentCocoa.mm:
(API::Attachment::setFileWrapperAndUpdateContentType):
* Source/WebKit/UIProcess/API/Cocoa/WKNavigationResponse.mm:
(-[WKNavigationResponse _downloadAttribute]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommand.mm:
(-[WKWebExtensionCommand debugDescription]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView mediaType]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewTesting.mm:
(-[WKWebView _contentsOfUserInterfaceItem:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
(+[WKWebsiteDataStore _removeDataStoreWithIdentifier:completionHandler:]):
(-[WKWebsiteDataStore _getAllBackgroundFetchIdentifiers:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm:
(nullableNSString):
(-[_WKApplicationManifest rawJSON]):
(-[_WKApplicationManifest name]):
(-[_WKApplicationManifest shortName]):
(-[_WKApplicationManifest applicationDescription]):
* Source/WebKit/UIProcess/API/Cocoa/_WKNotificationData.mm:
(-[_WKNotificationData title]):
(-[_WKNotificationData lang]):
(-[_WKNotificationData body]):
(-[_WKNotificationData tag]):
(-[_WKNotificationData origin]):
(-[_WKNotificationData identifier]):
* Source/WebKit/UIProcess/API/Cocoa/_WKTargetedElementInfo.mm:
(-[_WKTargetedElementInfo debugDescription]):
* Source/WebKit/UIProcess/Cocoa/SOAuthorization/SOAuthorizationSession.mm:
(WebKit::SOAuthorizationSession::continueStartAfterDecidePolicy):
* Source/WebKit/UIProcess/Cocoa/WKShareSheet.mm:
(-[WKShareSheet presentWithParameters:inRect:completionHandler:]):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionControllerAPITestCocoa.mm:
(WebKit::WebExtensionController::testResult):
(WebKit::WebExtensionController::testEqual):
(WebKit::WebExtensionController::testLogMessage):
(WebKit::WebExtensionController::testSentMessage):
(WebKit::WebExtensionController::testAdded):
(WebKit::WebExtensionController::testStarted):
(WebKit::WebExtensionController::testFinished):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCommandCocoa.mm:
(WebKit::WebExtensionCommand::setActivationKey):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView selectedText]):
* Source/WebKit/UIProcess/ios/WKPDFView.mm:
(-[WKPDFView actionSheetAssistant:performAction:]):
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
(-[WKFullScreenViewController setLocation:]):
(-[WKFullScreenViewController loadView]):
(-[WKFullScreenViewController _showPhishingAlert]):
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::registerEditCommand):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPICommandsCocoa.mm:
(WebKit::WebExtensionContextProxy::dispatchCommandsCommandEvent):
(WebKit::WebExtensionContextProxy::dispatchCommandsChangedEvent):
* Source/WebKit/webpushd/WebClipCache.mm:
(WebPushD::WebClipCache::persist):
* Source/WebKitLegacy/ios/WebView/WebPDFViewIOS.mm:
(-[WebPDFView _checkPDFTitle]):
* Source/WebKitLegacy/ios/WebView/WebPDFViewPlaceholder.mm:
(-[WebPDFViewPlaceholder _updateTitleForDocumentIfAvailable]):
* Source/WebKitLegacy/mac/Misc/WebNSURLExtras.mm:
(-[NSString _web_decodeHostName]):
(-[NSString _web_encodeHostName]):
(-[NSString _webkit_decodeHostName]):
(-[NSString _webkit_encodeHostName]):
* Tools/TestWebKitAPI/Tests/WTF/cocoa/URLExtras.mm:
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_ParsingError)):

Canonical link: https://commits.webkit.org/293429@main
